src/main/java/com/iexec/worker/result/ResultService.java

```diff
@@ -297,25 +302,22 @@ String buildResultLink(String storage, String location) {
     public String getIexecUploadToken() {
         // get challenge
         Integer chainId = blockchainAdapterConfigurationService.getChainId();
-        Optional<Eip712Challenge> oEip712Challenge = customResultFeignClient.getResultChallenge(chainId);
-
-        if (oEip712Challenge.isEmpty()) {
-            return "";
-        }
-
-        Eip712Challenge eip712Challenge = oEip712Challenge.get();
+        try {
+            Eip712Challenge eip712Challenge = resultProxyClient.getChallenge(chainId);
```

Comment:
Any chance this could be `null`? Shouldn't we log some thing in that case?

Reply:
If it happens, it will produce a `NullPointerException` which will be send us to the catch block (`NullPointerException` is a `RuntimeException`).
We can add a `log.error` statement to trace the exception origin there.